src/InstallationValidator.Core/Presentation/SimulationComparisonPresenter.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using InstallationValidator.Core.Assets;
+using InstallationValidator.Core.Events;
+using InstallationValidator.Core.Presentation.DTO;
+using InstallationValidator.Core.Presentation.Views;
+using InstallationValidator.Core.Services;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Core.Services;
+using OSPSuite.Presentation.Presenters;
+using OSPSuite.Utility.Events;
+
+namespace InstallationValidator.Core.Presentation
+{
+
+   public interface ISimulationComparisonPresenter : IDisposablePresenter,
+      IListener<AppendTextToLogEvent>,
+      IListener<AppendLineToLogEvent>
+   {
+      Task StartComparison();
+      void Abort();
+      void SelectFirstFolder();
+      void SelectSecondFolder();
+   }
+
+   public class SimulationComparisonPresenter : AbstractDisposablePresenter<ISimulationComparisonView, ISimulationComparisonPresenter>, ISimulationComparisonPresenter
+   {
+      private readonly FolderDTO _firstFolderDTO = new FolderDTO();
+      private readonly FolderDTO _secondFolderDTO = new FolderDTO();
+      private readonly IInstallationValidatorConfiguration _configuration;
+      private readonly IDialogCreator _dialogCreator;
+      private CancellationTokenSource _cancellationTokenSource;
+      private bool _comparisonRunning;
+      private readonly IBatchComparisonTask _batchComparisonTask;
+      private readonly IBatchComparisonReportingTask _batchComparisonReportingTask;
+
+      public SimulationComparisonPresenter(ISimulationComparisonView view, IInstallationValidatorConfiguration configuration, IDialogCreator dialogCreator, 
+         IBatchComparisonTask batchComparisonTask, IBatchComparisonReportingTask batchComparisonReportingTask) : base(view)
+      {
+         _configuration = configuration;
+         _dialogCreator = dialogCreator;
+         _batchComparisonTask = batchComparisonTask;
+         _batchComparisonReportingTask = batchComparisonReportingTask;
+         _secondFolderDTO.FolderPath = configuration.BatchOutputsFolderPath;
+         view.BindTo(_firstFolderDTO, _secondFolderDTO);
+      }
+
+      private void logText(string textToLog, bool isHtml = true)
```

Comment:
I think there is an easy way to avoid duplication simply be creating an interface implementing the 2 IListener<xx> and also a few extension methods on that interface. That way you avoid the code duplication AND the base class creation